lib/resty/mlcache.lua

```diff
@@ -665,10 +665,15 @@ local function check_opts(self, opts)
 end
 
 
-local function unlock_and_ret(lock, res, err, hit_lvl)
+local function unlock_and_ret(lock, res, err, hit_lvl, key)
```

Comment:
This new argument might not be necessary. lua-resty-lock objects have `self.key` once `:lock()` is called: https://github.com/openresty/lua-resty-lock/blob/master/lib/resty/lock.lua#L161 - let's try to leverage it instead.